Parser.cpp

```diff
@@ -146,33 +242,30 @@ std::optional<std::string> commonItems::parser::getNextToken(std::istream& theSt
 
 		auto matched = tryToMatchAgainstKeywords(toReturn, strippedLexeme, isLexemeQuoted, theStream);
 
-		if (!matched)
-		{
-			for (const auto& [matcher, parsingFunction]: registeredMatchers)
-			{
-				if (matcher(toReturn))
-				{
-					parsingFunction(toReturn, theStream);
+		// regexes and matchers
+		if (!matched) {
+			auto matchCaller = [&toReturn, &theStream](auto& obj) {
```

Comment:
Can this be made a completely freestanding function that takes everything as parameters, so that it doesn't need to be created for each call to getNextToken()?

Reply:
For that we can replace the lambda with a struct: https://www.bfilipek.com/2020/04/variant-virtual-polymorphism.html#passing-arguments

Reply:
I see. You might try using structured bindings to create one on-the-fly (based on an existing definition).

Though maybe its not a big deal. Have you tried this (or the polymorphism form) against an actual converter? What's the performance like?

Reply:
> I see. You might try using structured bindings to create one on-the-fly (based on an existing definition).

Do you mean using brace initialization inside the std::visit like in the example?
```
std::visit(CallPrintName{"intro text"}, var);
```

> Though maybe its not a big deal. Have you tried this (or the polymorphism form) against an actual converter? What's the performance like?

No, but I can do a comparison in any converter other than ImpToCK3 (because in this one I already updated commonItems to a version with the wrong ordering problem).

Reply:
Ah, yes, braced initialization. I don't know why I always get the terms confused.